beacon-chain/core/electra/deposits.go

```diff
@@ -81,7 +250,44 @@ func ProcessPendingBalanceDeposits(ctx context.Context, st state.BeaconState, ac
 func ProcessDepositRequests(ctx context.Context, beaconState state.BeaconState, requests []*enginev1.DepositRequest) (state.BeaconState, error) {
 	_, span := trace.StartSpan(ctx, "electra.ProcessDepositRequests")
 	defer span.End()
-	// TODO: replace with 6110 logic
-	// return b.ProcessDepositRequests(beaconState, requests)
+	var err error
+	for _, receipt := range requests {
+		beaconState, err = processDepositRequest(beaconState, receipt)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not apply deposit receipt")
+		}
+	}
 	return beaconState, nil
 }
+
+// processDepositRequest processes the specific deposit receipt
+// def process_deposit_request(state: BeaconState, deposit_request: DepositRequest) -> None:
+//
+//	# Set deposit request start index
+//	if state.deposit_requests_start_index == UNSET_DEPOSIT_REQUEST_START_INDEX:
+//	    state.deposit_requests_start_index = deposit_request.index
+//
+//	apply_deposit(
+//	    state=state,
+//	    pubkey=deposit_request.pubkey,
+//	    withdrawal_credentials=deposit_request.withdrawal_credentials,
+//	    amount=deposit_request.amount,
+//	    signature=deposit_request.signature,
+//	)
+func processDepositRequest(beaconState state.BeaconState, request *enginev1.DepositRequest) (state.BeaconState, error) {
+	receiptsStartIndex, err := beaconState.DepositRequestsStartIndex()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get deposit requests start index")
+	}
+	if receiptsStartIndex == params.BeaconConfig().UnsetDepositRequestsStartIndex {
+		if err := beaconState.SetDepositRequestsStartIndex(request.Index); err != nil {
+			return nil, errors.Wrap(err, "could not set deposit requests start index")
+		}
+	}
+	return ApplyDeposit(beaconState, &ethpb.Deposit_Data{
+		PublicKey:             bytesutil.SafeCopyBytes(request.Pubkey),
+		Amount:                request.Amount,
+		WithdrawalCredentials: bytesutil.SafeCopyBytes(request.WithdrawalCredentials),
+		Signature:             bytesutil.SafeCopyBytes(request.Signature),
+	}, true) // individually verify signatures instead of batch verify
```

Comment:
we may need to revisit this for operational efficiency especially if it's part of the transition_no_verifiy_sig ( it verifies) . instead we can add a verification  as a batch here.

Reply:
decided handle it in this PR